OTAnalytics/plugin_prototypes/track_visualization/track_viz.py

```diff
@@ -519,8 +518,16 @@ def _plot_dataframe(self, track_df: DataFrame, axes: Axes) -> None:
             alpha=self._alpha,
             ax=axes,
             palette=self._color_palette_provider.get(),
-            legend=self._enable_legend,
         )
+        if self._enable_legend:
```

Comment:
Where did self._enable_legend come from? Is this for a future feature to toggle the legend?

Reply:
`_enable_legend` is a member variable of `TrackGeometryPlotter`. In `main_application.py` several instances of `TrackGeometryPlotter` are created and passed to several layers that want to plot tracks on the canvas. Thus, we don't differentiate between plotters used in the different layers using it. Hence, the need to for `_enable_legend`. If you have a look at the `_create_layers` method in `main_application.py` you can see how plotters are being configured.

In the future, the member variable will be not necessary anymore, if the legend is displayed outside of the canvas.